### ISSUE TITLE
rke.cattle.io/external-ip is not updated when EC2 public IP is changed

### ISSUE THREAD

**Issue by PeterMaco** [rancher/rancher#27249]
**What kind of request is this (question/bug/enhancement/feature request):**
This is a bug

**Steps to reproduce (least amount of steps as possible):**
In RKE cluster in AWS stop and then start EC2 instance that is a cluster worker node with a public IP.

**Result:**
AWS assigns new public IP to the instance. Rancher shows correct new external IP in nodes view, but annotation rke.cattle.io/external-ip is not updated and contains old IP. This causes an issue to external DNS service which updates DNS to old IPs and it is not possible to connect to cluster.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
2.3.6
- Installation option (single install/HA):
Single install

**Cluster information**
- RKE Cluster hosted in AWS

- Kubernetes version

```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.1", GitCommit:"d647ddbd755faf07169599a625faf302ffc34458", GitTreeState:"clean", BuildDate:"2019-10-02T17:01:15Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4", GitCommit:"8d8aa39598534325ad77120c120a22b3a990b5ea", GitTreeState:"clean", BuildDate:"2020-03-12T20:55:23Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version 

```
Client:
 Version:      1.10.3
 API version:  1.24
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 21:49:11 2016
 OS/Arch:      linux/amd64
```

**Comment by wilsonge** [comment]
Duplicate of https://github.com/rancher/rancher/issues/14102 although 👍 for fixing

**Comment by dlednik** [comment]
FYI:

Also see this in v2.4.4

I restarted machine with rancher server which did not help. But when I manually restarted the EC2 server the cluster in operational again in UI. Note that the IP in my case was changed on a server that has etcd, control plane + worker and this is the only etcd atm in the cluster.

One `issue` that still remains that the IP in the list of nodes is still recorder as the old one.
